frontend/src/app/transactions/page.tsx

```diff
@@ -1,18 +1,218 @@
+            {/* Custom Arrow Icon */}
+            <div className="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2">
+              <svg
+                className="h-5 w-5 text-gray-400 dark:text-gray-300"
+                xmlns="http://www.w3.org/2000/svg"
+                viewBox="0 0 20 20"
+                fill="currentColor"
+                aria-hidden="true"
+              >
+                <path
+                  fillRule="evenodd"
+                  d="M10 12a1 1 0 01-.7-.3l-3-3a1 1 0 111.4-1.4L10 9.586l2.3-2.3a1 1 0 111.4 1.4l-3 3A1 1 0 0110 12z"
+                  clipRule="evenodd"
+                />
+              </svg>
+            </div>
+          </div>
+
+          {/* Company Name Filter */}
+          <div className="relative w-full sm:w-1/2">
+            <label htmlFor="company" className="block text-sm font-semibold text-gray-700 dark:text-gray-200 mb-1">
+              Company
+            </label>
+            <input
+              type="text"
+              id="company"
+              value={filters.company}
+              onChange={(e) => updateFilter('company', e.target.value)}
+              placeholder="Search by company name"
+              className="block w-full rounded-md border border-stroke bg-white px-3 py-2 shadow-default focus:border-green-500 focus:outline-none focus:ring-2 focus:ring-green-500 dark:bg-boxdark dark:border-strokedark dark:text-gray-200"
+            />
+          </div>
+        </div>
+
+        {/* Date Range Picker */}
+        <div className="relative w-full sm:w-1/3">
+          <label htmlFor="dateRange" className="block text-sm font-semibold text-gray-700 dark:text-gray-200 mb-1">
+            Date Range
+          </label>
+          <div className="relative overflow-visible z-50">
+          <Datepicker
+              primaryColor={"green"}
+              asSingle={false}
+              value={filters.dateRange}
+              onChange={handleDateChange}
+              showShortcuts={true}
+              placeholder="Select date range"
+              displayFormat="DD/MM/YYYY"
+              containerClassName="relative w-full rounded-md border border-stroke bg-white dark:bg-boxdark px-3 py-2 shadow-default focus-within:border-green-500 focus-within:ring-2 focus:ring-green-500 flex items-center h-10"
+              inputClassName="block w-full bg-transparent border-none focus:outline-none dark:text-gray-200 h-full leading-normal"
+            />
+          </div>
+        </div>
+      </div>
+      {/* End Filter Section */}
+
+      {/* Transactions Table */}
+      {loading && (
+        <div className="flex items-center justify-center">
+          <p className="text-lg text-gray-600">Loading transactions...</p>
+        </div>
+      )}
+
+      {error && (
+        <div className="flex items-center justify-center">
+          <p className="text-lg text-red-600">{error}</p>
+        </div>
+      )}
+
+      {!loading && !error && (
```

Comment:
In cases where the page fails to load the transactions, the page looks like this:

<img width="1018" alt="Screen Shot 2024-11-24 at 1 05 11 PM" src="https://github.com/user-attachments/assets/3e791e77-c171-4e17-bd60-37126af461cd">

Which is misleading because it says "displaying mock data" but nothing is displayed. To fix this, you could replace `!loading && !error` with `!loading`, so that the displaying doesn't depend on there being no error. This would yield the following:

<img width="1031" alt="Screen Shot 2024-11-24 at 1 16 41 PM" src="https://github.com/user-attachments/assets/095a883b-590d-4416-b599-32803f18cb3a">

(I've also slightly changed the text and padding, and added an icon to the left)

Let me know if this makes sense, and I can commit it if so.

Reply:
@sonofthomp Thanks Gabriel! That's amazing feedback I fully agree. Let's go with these new changes. You can commit and merge with your new changes.